_includes/source_code/code/34-PDP/smalltank/smalltank_disjoint_sets.cc

```diff
@@ -0,0 +1,73 @@
+#include <algorithm>
+#include <cstdio>
+
+typedef std::tuple<long /* βάρος */, long /* 1η κορυφή */, long /* 2η κορυφή */> Edge;
+
+const long MAXN = 10000;
+
+int par[MAXN + 1];
+int sz[MAXN + 1];
+
+/* Αρχικοποίηση των disjoint sets. */
+void init(long N) {
+   for (int i = 1; i <= N; ++i) {
+      par[i] = i;
+      sz[i] = 1;
+   }
+}
+
+/* Βρίσκουμε τον αντιπρόσωπο του συνόλου. */
+long find_parent(long u) {
+   if (u == par[u]) return u;
+   return par[u] = find_parent(par[u]);
+}
+
+/* Ενώνουμε τα σύνολα που περιέχουν το 
+   u και το v και επιστρέφουμε αν έγινε 
+   καινούργια ένωση. */
+bool merge(long u, long v) {
+   long p_u = find_parent(u);
+   long p_v = find_parent(v);
+   
+   if (p_u == p_v) return false;
+   
+   if (sz[p_u] < sz[p_v]) par[p_u] = p_v;
```

Comment:
προτείνω αν το ένα σκέλος της if έχει άγγιστρα, να έχει και το άλλο. Εφόσον συχνά χρησιμοποιείται π.χ. for με περιττά άγγιστρα για μια εντολή, ας είναι ομοιόμορφη η if και ας έχουμε μια γραμμή χαμμένη από το {

Reply:
Ναι, καλύτερα έτσι